File: src/Week237/T1.java
package Week237;

public class T1 {
    public boolean checkIfPangram(String sentence) {
        int[] arr = new int[26];
        for (int i = 0; i < sentence.length(); i++) {
            arr[sentence.charAt(i) - 97]++;
        }
        for (int i : arr) {
            if (i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: src/Week264/T1.js
/**
 * @param {string} sentence
 * @return {number}
 */
var countValidWords = function(sentence) {
    let arr = sentence.split(' ');
    let reg = /^([!.,]|[a-z]+-[a-z][!.,]?|[a-z]+[!.,]?)$/;
    let ret = 0;
    for (let s of arr) {
        if (s.match(reg)) {
            ret++;
        }
    }
    return ret;
};

countValidWords("he bought 2 pencils, 3 erasers, and 1  pencil-sharpener.")